src/config.js

```diff
@@ -1,20 +1,120 @@
+ * @param  {"Custom"|"126"|"163"|"1und1"|"AOL"|"DebugMail"|"DynectEmail"|"FastMail"|"GandiMail"|"Gmail"|"Godaddy"|"GodaddyAsia"|"GodaddyEurope"|"hot.ee"|"Hotmail"|"iCloud"|"mail.ee"|"Mail.ru"|"Maildev"|"Mailgun"|"Mailjet"|"Mailosaur"|"Mandrill"|"Naver"|"OpenMailBox"|"Outlook365"|"Postmark"|"QQ"|"QQex"|"SendCloud"|"SendGrid"|"SendinBlue"|"SendPulse"|"SES"|"SES-US-EAST-1"|"SES-US-WEST-2"|"SES-EU-WEST-1"|"Sparkpost"|"Yahoo"|"Yandex"|"Zoho"|"qiye.aliyun"} userConfig.mailService- Set mailing service to use defaults to gmail
+ * @param  {String} userConfig.senderEmail set sender's email in case of supported services or user in case of custom service
+ * @param  {String} userConfig.senderPassword set sender's password in case of supported services or user in case of custom service
+ * @param  {String} [userConfig.host] - Host for your custom service
+ * @param  {Number} [userConfig.port] - Port for your custom service
+ * @param  {Boolean} [userConfig.secure ]- Set to true for your custom service if you want a (TLS) secured connection
+ * @throws Invalid Configurations provided for custom service
+ * @throws Invalid settings provided for senderEmail, senderPasword, or mailService
+ * @returns {config} configuration that has been set
+ */
+function setConfig(
+  userConfig = {
+    hostingURL: '',
+    mailService: 'Gmail',
+    senderEmail: '',
+    senderPassword: '',
+    host: '',
+    port: 465,
+    secure: false,
+  }
+) {
+  config = {}
+  // config.hostingURL = (validator.isURL(userConfig.hostingURL || process.env.HOSTING_URL))?userConfig.hostingURL || process.env.HOSTING_URL: '';
+  config.hostingURL = userConfig.hostingURL || process.env.HOSTING_URL;
+  // DONE: Better  validation
+  
+  if ((userConfig.mailService || process.env.MAIL_SERVICE) === 'Custom') {
+    if ( validator.isFQDN(userConfig.host || process.env.MAIL_HOST) && validator.isPort(`${userConfig.port || process.env.MAIL_PORT}`) && !validator.isEmpty(userConfig.senderEmail || process.env.MAIL_USER) && !validator.isEmpty(userConfig.senderPassword || process.env.MAIL_PASSWORD) && validator.isBoolean(`${userConfig.secure || process.env.MAIL_SECURE}`) ) {
+      config.mailService = 'Custom'
+      config.host = userConfig.host || process.env.MAIL_HOST;
+      config.port = userConfig.port || process.env.MAIL_PORT;
+      config.secure = userConfig.secure || process.env.MAIL_SECURE;
+      config.auth = {
+        user: userConfig.senderEmail || process.env.MAIL_USER,
+        pass: userConfig.senderPassword || process.env.MAIL_PASSWORD,
+      };
+      return config
+    } else {
+      throw new Error('Invalid Configurations provided for custom service');
+    }
+  }
+  config.mailService = (userConfig.mailService || process.env.MAIL_SERVICE)? userConfig.mailService || process.env.MAIL_SERVICE : 'Gmail';
+  if (validator.isEmail(userConfig.senderEmail || process.env.MAIL_USER) && !validator.isEmpty(userConfig.senderPassword || process.env.MAIL_PASSWORD)) {
```

Comment:
Split this into individual checks and individual errors. For example, if `senderEmail` not set, throw -> "senderEmail not set in the config".